Xamarin.Essentials/Calendar/Calendar.ios.cs

```diff
@@ -27,16 +27,48 @@ static async Task<IReadOnlyList<ICalendar>> PlatformGetCalendarsAsync()
             return calendarList.AsReadOnly();
         }
 
-        static Task<IReadOnlyList<IEvent>> PlatformGetEventsAsync(string calendarId = null, DateTimeOffset? startDate = null, DateTimeOffset? endDate = null) => throw new NotImplementedException();
-
-        static async Task PlatformRequestCalendarReadAccess()
+        static async Task<IReadOnlyList<IEvent>> PlatformGetEventsAsync(string calendarId = null, DateTimeOffset? startDate = null, DateTimeOffset? endDate = null)
         {
             await Permissions.RequireAsync(PermissionType.CalendarRead);
-        }
 
-        static async Task PlatformRequestCalendarWriteAccess()
-        {
-            await Permissions.RequireAsync(PermissionType.CalendarWrite);
+            var systemAbsoluteReferenceDate = new DateTime(2001, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+            var eventList = new List<Event>();
+            var startDateToConvert = startDate ?? DateTimeOffset.Now;
+            var endDateToConvert = endDate ?? startDateToConvert.AddYears(4);   // 4 years is the maximum period that a iOS calendar events can search
+            var sDate = NSDate.FromTimeIntervalSinceReferenceDate((startDateToConvert.UtcDateTime - systemAbsoluteReferenceDate).TotalSeconds);
+            var eDate = NSDate.FromTimeIntervalSinceReferenceDate((endDateToConvert.UtcDateTime - systemAbsoluteReferenceDate).TotalSeconds);
+            var calendars = !string.IsNullOrEmpty(calendarId) ? CalendarRequest.Instance.Calendars.Where(x => x.CalendarIdentifier == calendarId).ToArray() : null;
+            var query = CalendarRequest.Instance.PredicateForEvents(sDate, eDate, calendars);
+            var events = CalendarRequest.Instance.EventsMatching(query);
+
+            foreach (var e in events)
+            {
+                eventList.Add(new Event
+                {
+                    Id = e.CalendarItemIdentifier,
+                    Title = e.Title,
+                    Start = (long)Math.Floor((Math.Abs(NSDate.FromTimeIntervalSince1970(0).SecondsSinceReferenceDate) + e.StartDate.SecondsSinceReferenceDate) * 1000),
```

Comment:
Should we create a extension/helper method for the unix date calculation?

Reply:
Added in latest build: 

Preview:
public static class CalendarExtensions
{
    static DateTime IosNSDateTimeSystemZeroPoint = new DateTime(2001, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);

    public static long ToEpochTime(this NSDate date) => (long)Math.Floor((Math.Abs(NSDate.FromTimeIntervalSince1970(0).SecondsSinceReferenceDate) + date.SecondsSinceReferenceDate) * 1000);

    public static NSDate ToNSDate(this DateTimeOffset date) => NSDate.FromTimeIntervalSinceReferenceDate((date.UtcDateTime - IosNSDateTimeSystemZeroPoint).TotalSeconds);
}

Reply:
👍